src/app/search/page.tsx

```diff
@@ -13,118 +15,124 @@ import {
+
+  // 데이터 유효성 검사
+  const isValidData = !(isLoading || searchProjects.length === 0 || isError);
 
-  const totalItems = 30; // 총 프로젝트가 30개라고 가정
-  const itemsPerPage = 8; // 한 페이지에 8개씩 보여줌
-  const totalPages = Math.ceil(totalItems / itemsPerPage); // 4
+  // 페이징 관련
+  const totalItems = data?.data.totalCount || DEFAULT_ITEMS_TOTAL_COUNT;
+  const totalPages = Math.ceil(totalItems / ITEMS_PER_PAGE);
 
   const handleClickPaginationItem = (page: number) => {
     setCurrentPage(page);
   };
-  const testData = [
-    {
-      projectId: 1,
-      projectname: 'project.projectName1',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 2,
-      projectname: 'project.proj4ectName2',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 3,
-      projectname: 'project.project1Name',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 5,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 6,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 7,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 8,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-  ];
+
+  // "이전" 버튼
+  const handlePrevious = () => {
+    if (currentPage > 1) {
+      setCurrentPage(currentPage - 1);
+    } else {
+      setCurrentPage(0);
+    }
+  };
+
+  // "다음" 버튼
+  const handleNext = () => {
+    if (currentPage < totalPages - 1) {
+      setCurrentPage(currentPage + 1);
```

Comment:
여기도 현재 페이지가 마지막 페이지일 경우에도 페이지를 totalPages - 1로 설정하는 것 같은데, currentPage가 totalPages - 1보다 작은 경우에만 페이지를 증가시키는 것이 명확할 것 같아요
```suggestion
      setCurrentPage((prevPage) => prevPage + 1);
  }
};
```

```
const handleNext = () => {
  if (currentPage < totalPages - 1) {
    setCurrentPage((prevPage) => prevPage + 1);
  }
};
```

Reply:
✅ 0e5b96ceda8262219f047f3878ce120d22648439